src/Our.Umbraco.AuthU/Web/Controllers/OAuthController.cs

```diff
@@ -76,6 +77,25 @@ protected void ProcessClient(OAuthTokenRequest request)
                 Client = client;
             }
         }
+	
+	protected void SetOriginHeader()
+	{
+		string allowedOrigin = Client != null ? Client.AllowedOrigin : Context.Options.AllowedOrigin;
+
+		if (HttpContext.Current.Response.Headers.AllKeys.Contains("Access-Control-Allow-Origin"))
+		{
+			var accessControlHeader = HttpContext.Current.Response.Headers.GetValues("Access-Control-Allow-Origin").FirstOrDefault();
+
+			if (!accessControlHeader.Equals(allowedOrigin, StringComparison.OrdinalIgnoreCase))
+			{
+				throw new Exception($"There is currently a header set with the key Access-Control-Allow-Origin, but the value: {accessControlHeader} differs from the OAuth configured value: {allowedOrigin}");
```

Comment:
Could we look at throwing a `OAuthResponseException` instead of `Exception` so it's something that can be read from the app? I'd probably use the `HttpStatusCode.InternalServerError` status code and have a msg body like:

    new { invalid_allowed_origin = "INSERT_MSG_HERE" }

This way it's just consistent with how the rest of the API throws errors.